CRUD operations for Sequence objects + add/remove page elements

Deep-Chill opened the pull request:
`elements.py`:
- Added SequenceAPIView to list, create, update, or delete Sequence objects.
- Includes add_element and remove_element custom actions to add/remove EnumeratedElements.

`serializers.py`:
- UpdateEnumeratedElementSerializer: Serializer for updating enumeratedelements, where page_element is a queryset of all available page_elements.
- SequenceSerializer with elements listed for the sequence.

Deep-Chill committed:
CRUD operations for Sequence objects + add/remove elements belonging to a sequence

Deep-Chill committed:
Much simpler way to get the user's "username" and enumerated elements' slugs

Added new serializers: One for listing EnumeratedElements, another for updating Sequence. Simplified existing serializers to display the usernames and slugs more gracefully.

There is still a discrepancy, though, the API expects the user's ID instead of their username for post/update requests. Maybe we should just use the ID everywhere? That would mean we get rid of the UpdateSequenceSerializer but that would cause it to show users as IDs, while they are showing as usernames in existing serializers(where they're read_only) ex. NodeElementSerializer

Could also use
"    account = serializers.SlugRelatedField(
        required=False,
        queryset=get_user_model().objects.all(),
        slug_field=settings.ACCOUNT_LOOKUP_FIELD,
        help_text=_("Account the sequence belongs to"))
"

smirolo commented on pages/api/elements.py:
```diff
@@ -645,3 +647,223 @@ def perform_update(self, serializer):
+            ]
+        }
+    ]
+    - **Create a Sequence**
```
Split API docs, one block per HTTP method to comply with expectations from docgen tools.

smirolo commented on pages/api/elements.py:
```diff
@@ -645,3 +647,223 @@ def perform_update(self, serializer):
+
+    .. code-block:: http
+
+        GET /api/sequences/ HTTP/1.1
```
API URL should end without a '/' per convention. i.e `/api/sequences` instead of `/api/sequences/`.

smirolo commented on pages/api/elements.py:
```diff
@@ -645,3 +647,223 @@ def perform_update(self, serializer):
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        self.perform_destroy(instance)
+        return api_response.Response({'detail': 'sequence deleted'}, status=200)
```
DELETE typically returns "204 NO Content" as in `rest_framework.mixins.DestroyModelMixin`

smirolo commented on pages/serializers.py:
```diff
@@ -261,3 +261,65 @@ class PageElementTagSerializer(serializers.ModelSerializer):
+    Serializes an EnumeratedElement
+    """
+
+    page_element = serializers.SerializerMethodField()
```
OK. As an idea, `SlugField(source='page_element.slug')` might also work instead of defining `ger_page_element`.

smirolo commented on pages/api/elements.py:
```diff
@@ -645,3 +647,223 @@ def perform_update(self, serializer):
+
+    .. code-block:: http
+
+        PUT /api/sequences/1/ HTTP/1.1
```
The API should be defined using the `slug` of the sequence instead of its `id` in the database to prevent "id walking" - i.e. iterating the database elements because of a predictable pattern to identify sequences.

smirolo reviewed (changes requested):
Use `slug` of the sequence instead of database serial primary key.

Split API documentation in 3 blocks for get, post and delete.

Deep-Chill committed:
Refactoring

Added documentation within each method

Set the url to not use trailing slashes.

Removed updateserializers

Use slug instead of IDs

Fixed return code

smirolo reviewed (approved):
Good start to enable online courses within the app.
